### ISSUE TITLE
Support for List/Array types 

### ISSUE THREAD

**Issue by flosell** [evidence-dev/evidence#1558]
## Feature Description

DuckDB supports a [LIST type](https://duckdb.org/docs/sql/data_types/list) (among other [nested data types](https://duckdb.org/docs/sql/data_types/overview#nested--composite-types)) yet seems evidence at the moment only supports very [basic](https://github.com/evidence-dev/evidence/blob/main/packages/db-commons/index.cjs#L6) data-types, [defaulting to string](https://github.com/evidence-dev/evidence/blob/main/packages/duckdb/index.cjs#L44) for all others. 

I propose adding support for DuckDBs nested data types, starting with LIST.

## Goal of Feature

Supporting these data types would mean being able to use more of DuckDBs power when modelling and querying data. It would also reduce confusion (e.g. [recently on Slack](https://evidencedev.slack.com/archives/C023LRB9Z40/p1705921193843869)) when columns silently change type. 

## Current Solution / Workarounds

Seems like the full power of DuckDB is available in DuckDB [source queries](https://docs.evidence.dev/core-concepts/data-sources/#configure-source-queries) so at the moment, I do all work requiring the list type there. 

There's no live reloading with source queries though. Also, having to keep in mind which kind of query I'm writing and where to put it adds significant cognitive load. 
Long story short, the workaround is alright but makes for bad developer experience. 

## Other notes

I'm happy to help with contributions myself, however seems like this might be a more complex change so might need some pointers how to approach it.

**Comment by justin0mcateer** [comment]
I have run into this same problem, our historical data has numerous nested structures, as our intention has been to query it with DuckDB from the start, the use of DuckDB is what attracted us to Evidence.

I see in the 'static/data' folder there is a '*.schema.json' file, which is being generated along with the parquet file when 'npm run sources' is being executed. All of the columns with structured data are being output as strings (JSON), the schema shows these as strings.

I would assume that the structures could be reconstituted and queried using the JSON functions of DuckDB. However, this would likely be much slower than the native structures which DuckDB and Parquet both support. Also, this will require overriding the the duckdb-wasm build which is being used.

**Comment by paulmech** [comment]
just adding my 2 cents here as well - i would also benefit from this feature

in one of my use cases, a day's worth of data starts out as a 6gb collection of complex json files. i use python and duckdb to unnest the json files into around 10 separate tables, each having around 1.2 million rows.

a large amount of the data is repeated, and a technique i've used is to use window functions using the `array_agg` function to collate the "items of interest" in each table. the length of the array in my case is typically 24 elements. the array elements represent a time shift of 5 minutes; the first value is a recorded value, and all of the remaining values are what was forecast 5, 10, 15, 20, 25 mins etc. beforehand

my aim was to have 2 graphs side by side - left showing the recorded series (always element 1 of the array), and right showing one of the forecast series (e.g. a dropdown would be used to select one position of the array between positions 2-24)

**Comment by archiewood** [comment]
if you have this in your source as `my_query.sql`
```sql
select ['one','two','three'] as list_col
```

in your markdown file you can do
```sql
select list_value(list_col) 
from my_query
```

docs for [list_value](https://duckdb.org/docs/sql/functions/nested.html#list_valueany-)

**Comment by danielgafni** [comment]
Nothing really to add here but wanted to highlight the same problem with BigQuery. 
I have 2 array columns that I would like to load from BigQuery as is and then process with DuckDB (inside Evidence). 
Currently these columns are silently emptied when loading from BigQuery. 
Instead I had to write the following atrocity (unnesting in BigQuery seems to be way more complicated than in DuckDB): 

```sql
SELECT
    count_value AS count,
    breakpoint
FROM `${bq_dataset}.raw_video_stats_global`,
UNNEST(hist_counts) AS count_value WITH OFFSET AS idx
JOIN UNNEST(hist_breakpoints) AS breakpoint WITH OFFSET AS idx2
ON idx = idx2
WHERE property = 'duration'
```

As you can see this approach doesn't scale beyond a few columns.

**Comment by hughess** [comment]
Closing along with the release of the new Evidence Core: https://docs.evidence.dev/

Feel free to open a new issue if this is blocking you!
